torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -239,6 +239,21 @@ void annotateInputShapes(
   }
 }
 
+std::shared_ptr<Graph> removeUnusedSelfArgument(
+    const std::shared_ptr<Graph>& graph) {
+  if (graph->inputs().size() == 0) {
+    return graph;
+  }
+  jit::Value* self_argument = graph->inputs().at(0);
```

Comment:
Kind of a dumb question but how do we know this is a `self` argument?  Or is that kind of a red herring, and we just always try to remove an unused first argument (which happens to usually be self)?

Reply:
We check that the type of the argument is "module" in a check below, but overall it is possible that this argument is not `self`. The assumption here is that this pass will be used when having a self arguments prevents whole-graph compilation from happening, e.g.:
```
aot_artifact = torch.jit._C.compile_graph(graph)
# complains about unsupported type `something.something.Module`
graph2 = torch.jit._C.remove_unused_self(graph)
aot_artifact = torch.jit._C.compile_graph(graph2)
# succeeds
```